### ISSUE TITLE
Some questions about the flexibility of the library

### ISSUE THREAD

**Issue by Auties00** [avaje/avaje-jsonb#216]
Hello, I just found this library and I wanted to integrate it into one of my projects.
Though I have a couple of questions:
1. @Json.Value seems to work only on enums, unlike in Jackson where it works anywhere
Let's say I have an object that represents a String value, I can de-serialize the object from the String using @Json.Creator, but I can't serialize it as @Json.Value doesn't work on records and classes

2. Is there any way to make Optional accessors be recognized?
Let's say I have a model like this:
```
@Json
class Personal {
    private final String name;
    // Constructor
    public Optional<String> name() { ,,, }
}
```
The library will generate a piece of code that assumes the return type of the accessor matches the one of the property, so the following error is thrown by the compiler:
java: incompatible types: java.util.Optional<java.lang.String> cannot be converted to java.lang.String
Is there any way to fix this?

3. Json.Creator from map of attrributes
In Jackson you can do something like this:
```
record Something(SomeObject someObject) {
    @JsonCreator
    public static Something fromJson(Map<String, Object> json) {
        ...
    }
}
```
This doesn't appear to be possible in this library, is there any reason why? I saw that the library has an @Unmapped annotation, so this should be quite easy to implement assuming that if a JsonCreator exists that takes only a map all the fields should be considered unmapped

If these features sound reasonable, but are not implemented I can work on them and open matching PRs

**Comment by SentryMan** [comment]
1. we have an issue open for one, I'm just not fully sure how inlining should work. 
2. This is straightforward, just never occurred to me, gonna try and make a PR for this
3. Yeah could do this, problem is cases where people actually have a `Map<String, Object>`

**Comment by Auties00** [comment]
> 1. we have an issue open for one, I'm just not fully sure how inlining should work for more complex objects.
> 
>     2. This is straightforward, just never occurred to me, gonna try and make a PR for this
> 
>     3. Yeah could do this, problem is cases where people actually have a `Map<String, Object>` as part of the fields

I see that you opened a PR for 2 and 3. I'll see what I can do for issue number 1. Btw I saw someone asking for a real example in the other issue.
Let's say I have a string that represents an address:
`protocol://username:password@host:port/`
Then this string is a JSON object, but I want to parse that String as object. Then I can do
```
record Address(String protocol, String username, String password, String host, int port) {
    @Json.Creator
    // Some method that parses the string address as the OO representation
    
    @Json.Value
    // For example the toString method that reconverts the OO representation into the address 
}

```
Json.Creator already works like that, but @Json.Value doesn't

**Comment by SentryMan** [comment]
on item 2, what prevents you from doing

```java
@Json
class Personal {
    private final Optional<String> name;
    // Constructor
    public Optional<String> name() { ,,, }
} 
```

**Comment by Auties00** [comment]
> on item 2, what prevents you from doing
> 
> 
> 
> ```java
> 
> @Json
> 
> class Personal {
> 
>     private final Optional<String> name;
> 
>     // Constructor
> 
>     public Optional<String> name() { ,,, }
> 
> } 
> 
> ```
> 
> 

We don't have value types yet in Java, so it's an anti pattern to use optionals as field types. 
More detailed explanation:
https://blog.joda.org/2014/11/optional-in-java-se-8.html?m=1

Btw apart from this very specific use caee, there are cases where the accessor's type and and field type won't match for other reasons, so I think it's a nice feature to have.

**Comment by SentryMan** [comment]
>  where the accessor's type and and field type won't match for other reasons

Optionals I can understand, but this sounds sus to me. Despite that, I found out you can do this if you make an object like.

```java
@Json
public class OptionalAccess {
  @Ignore(deserialize = true)
  String stringy;

  @Property("stringy")
  public Optional<String> stringy() {
    return Optional.of(stringy);
  }

  public void stringy(String stringy) {
    this.stringy = stringy;
  }
}
```

**Comment by SentryMan** [comment]
all three PR's have been merged and uploaded to mvn central in 1.11-RC3
